### ISSUE TITLE
dataset in table 5

### ISSUE THREAD

**Issue by freedom6927** [TheShadow29/zsgnet-pytorch#13]
Hi,  could you release the dataset in Table5, thank you.

**Comment by TheShadow29** [comment]
@freedom6927 please see https://github.com/TheShadow29/zsgnet-pytorch/blob/master/DATA_README.md

You need to download the annotation files in the drive link. 

Let me know if you run into any error.

**Comment by freedom6927** [comment]
Hi, I have downloaded. But I can't distinguish which files are related to
table 5. I found all the files are related to table 2-4

Arka Sadhu ***@***.***> 于2022年2月11日周五 02:40写道：

**Comment by TheShadow29** [comment]
@freedom6927 it is under vg_split / csv_dir.

For training and validation, you could use `training_balanced.csv` and `val_balanced.csv`

For testing, you would use `test_balanced_c2.csv` and `test_balanced_c3.csv` respectively. 

In table 5, we train on train_balanced, and validate using `val_balanced and finally test it on `test_balanced_c2.csv` (VG-2B) and `test_balanced_c3.csv` (VG-3B)

For the distances, you would need the object name from `test_...csv` file and find the closest object in the training set and then simply bucket them into 3-4, 4-5 and so on, and get results for those each subset.

Let me know if that answers your question.

**Comment by freedom6927** [comment]
Got it, thanks!

Arka Sadhu ***@***.***> 于2022年2月12日周六 07:13写道：

**Comment by freedom6927** [comment]
Hi, I have a new question in table 4. are the datasets of
unblance test_c2.csv, test_c3.csv and train.csv. But where is the val.csv.
The only found of val is val_balanced.csv
Arka Sadhu ***@***.***> 于2022年2月12日周六 07:13写道：

**Comment by TheShadow29** [comment]
@freedom6927 sorry for the late reply, we only used the balanced validation set.

**Comment by freedom6927** [comment]
Ok, final question. Could you release the dataset division of Table 5 to
me? I hold refer to your designed ideas to finish my research.
Arka Sadhu ***@***.***> 于2022年2月15日周二 06:12写道：

**Comment by TheShadow29** [comment]
@freedom6927 Sorry, I don't understand your question. What do you mean by dataset division?

**Comment by freedom6927** [comment]
[image: table5.png]
you see. In table 5. The testset of vg2B and vg3B are divided into 5 parts
by using GloVe embeddings to compute the distances between test sample and
training sample. Could you release the results of division to  me (the
details of 310, 1050, 3543, 5321, 1985 of vg2B)

Arka Sadhu ***@***.***> 于2022年2月15日周二 11:52写道：

**Comment by TheShadow29** [comment]
@freedom6927 I don't have it with me, but something along the following lines of code should be sufficient:

```
train_df = ... # read train csv
test_df = ... # read test csv

train_objects = train_df['object_name'].unique()
test_objects = train_df['object_name'].unique()

glove_emb = ... # read glove embeddings

train_obj_emb = glove_emb(train_objects)
test_obj_emb = glove_emb(test_objects)

test_dist_dict = {}
for test_obj in test_objects:
    # find closest train object
    closest_train_obj = ....
    closest_train_obj_dist = ....
    test_dist_dict[test_obj] = closest_train_obj_dist

# bucket by distances

# compute scores for each object
```

Let me know if this answers your question.

**Comment by freedom6927** [comment]
thanks very much. I understand your idea

Arka Sadhu ***@***.***> 于2022年2月15日周二 12:35写道：
